src/compiler/checker.ts

```diff
@@ -37819,7 +37819,7 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
         if (getIsolatedModules(compilerOptions)) {
             Debug.assert(!!(type.symbol.flags & SymbolFlags.ConstEnum));
             const constEnumDeclaration = type.symbol.valueDeclaration as EnumDeclaration;
-            if (constEnumDeclaration.flags & NodeFlags.Ambient) {
+            if (constEnumDeclaration.flags & NodeFlags.Ambient && !(node.flags & NodeFlags.Ambient)) {

```

Comment:
I talked to @andrewbranch offline, and we think this change is correct but incomplete, meaning it doesn't capture all of the cases where it is actually safe to use an ambient const enum with `isolatedModules` on.
The idea is that as long as the access isn't going to be emitted, then it is safe and we shouldn't error. This is true if the access occurs in an ambient context, but it is also true in other contexts, e.g. in the original example, even if we're in a normal ts file, it should be ok to use an ambient const enum in a type.
A more complete way to detect when it's safe to access an ambient const enum is to use `isValidTypeOnlyAliasUseSite` from `compiler/utilities.ts`.

Reply:
@gabritto Thanks for the detailed feedback. I've made the change to use the `isValidTypeOnlyAliasUseSite` helper.